src/scripts/towns/DreamOrbController.ts

```diff
@@ -0,0 +1,123 @@
+            new DreamOrbLoot({type: ItemType.underground, id: 'Red Shard'}, 0.125),
+            new DreamOrbLoot({type: ItemType.underground, id: 'Blue Shard'}, 0.125),
+            new DreamOrbLoot({type: ItemType.underground, id: 'Yellow Shard'}, 0.125),
+            new DreamOrbLoot({type: ItemType.underground, id: 'Green Shard'}, 0.125),
+        ]),
+        new DreamOrb('Orange', new ObtainedPokemonRequirement(pokemonMap['Thundurus (Therian)']), [
+
+            new DreamOrbLoot({type: ItemType.item, id: 'Staryu'}, 0.133),
+            new DreamOrbLoot({type: ItemType.item, id: 'Ralts'}, 0.133),
+            new DreamOrbLoot({type: ItemType.item, id: 'Rotom'}, 0.134),
+            new DreamOrbLoot({type: ItemType.item, id: 'Landorus (Therian)'}, 0.075),
+            new DreamOrbLoot({type: ItemType.underground, id: 'Fire Stone'}, 0.25),
+            new DreamOrbLoot({type: ItemType.item, id: 'Protein'}, 0.275),
+        ]),
+        //new DreamOrb('Blue', new ObtainedPokemonRequirement(pokemonMap['Enamorus (Therian)']), [new DreamOrbLoot({type: ItemType.item, id: 'Enamorus (Therian)'}, 0.01)]),
+        //new DreamOrbLoot({type: ItemType.item, id: 'Igglybuff'}, 0.2),
+        //new DreamOrbLoot({type: ItemType.item, id: 'Smoochum'}, 0.2),
+        //new DreamOrbLoot({type: ItemType.item, id: 'Enamorus (Therian)'}, 0.075),
+        //new DreamOrbLoot({type: ItemType.underground, id: 'Moon Stone'}, 0.1),
+        //new DreamOrbLoot({type: ItemType.underground, id: 'Leaf Stone'}, 0.1),
+        //new DreamOrbLoot({type: ItemType.underground, id: 'Thunder Stone'}, 0.1),
+        //new DreamOrbLoot({type: ItemType.underground, id: 'Heart Scale'}, 0.125),
+        //new DreamOrbLoot({type: ItemType.underground, id: 'Max Revive'}, 0.1),
+    ]
+
+    public open() {
+        if (this.opening()) {
+            return;
+        }
+        if (!this.selectedOrb().amount) {
+            Notifier.notify({
+                message: 'No orbs left.',
+                type: NotificationConstants.NotificationOption.danger,
+            });
+            return;
+        }
+        this.opening(true);
+        this.item(undefined);
+        GameHelper.incrementObservable(this.selectedOrb().amount, -1);
+        const item = Rand.fromWeightedArray(this.selectedOrb().items, this.selectedOrb().items.map((i) => i.weight));
+        BagHandler.gainItem(item.item);
+        setTimeout(() => {
+            this.item(item);
+            this.opening(false);
+        }, 1800);
+    }
+
+    saveKey = 'dream-orbs';
+    defaults: Record<string, any>;
+    toJSON(): Record<string, any> {
+        return {
+            orbs: [
+                ...this.orbs.map((o) => {
+                    return {amount: o.amount(), color: o.color};
+                }),
+            ],
+        };
+    }
+    fromJSON(json: Record<string, any>): void {
+        json?.orbs?.forEach((o) => this.orbs.find((o2) => o2.color == o.color).amount(o.amount));
```

Comment:
Just in case we remove a color (spelling or name change maybe?)
```suggestion
        json?.orbs?.forEach((o) => this.orbs.find((o2) => o2.color == o.color)?.amount(o.amount));
```